openedx/core/djangoapps/discussions/api/providers.py

```diff
@@ -0,0 +1,29 @@
+from typing import List, Optional
+
+from opaque_keys.edx.keys import LearningContextKey
+
+from .config import get_discussion_config
+from ..discussions_apps import DiscussionApp, DiscussionAppsPluginManager
+
+
+def get_discussion_provider(context_key: LearningContextKey) -> Optional[DiscussionApp]:
+    """
+    Returns the discussion app provider associated with the provided context key.
+
+    Args:
+        context_key (LearningContextKey): Learning context, currently only a course
+
+    Returns:
+        A DiscussionApp instance for the discussion provider associated with this context.
+    """
+
+    config = get_discussion_config(context_key)
+    if config:
+        return DiscussionAppsPluginManager.get_plugin(config.provider)
+
+
+def get_available_discussion_providers() -> List[DiscussionApp]:
```

Comment:
This function doesn't seem to be used, far as I can tell.

Reply:
Currently, this is adding APIs that are designed for plugins to use, they may or may not be used within the platform. 

I will be creating a REST API that can be used to fetch some of this data so that it can be shown in MFE. That will be one user of this function.